File: Algorithms/Sorting/Selection Sort/SelectionSort.cpp
#include<iostream>
using namespace std;

int main(){
    int array[5] = {0,0,0,0,0};

    for(int i = 0; i<(sizeof(array)/sizeof(array[0])); i++){
        cin >> array[i]; 
    }

    cout << "Before: ";
    for(int i: array){
        cout << i << ',';
    }
    cout << endl;

    int min = 0;
    for(int i = 0; i<(sizeof(array)/sizeof(array[0])); i++){
        min = i;
        for(int j = i; j<(sizeof(array)/sizeof(array[0])); j++){
            if(array[min]>array[j]){
                min = j;
            }
        }
        if(i!=min){
            int temp = array[i];
            array[i] = array[min];
            array[min] = temp;
        }
    }

    cout << "After: ";
    for(int i: array){
        cout << i << ',';
    }
    cout << endl;

    return 0;
}

File: Algorithms/Sorting/Selection Sort/SelectionSort.java
import java.util.Scanner;

public class SelectionSort{
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int array[] = new int[5];
        for(int i = 0; i < array.length; i++){
            array[i] = sc.nextInt();
        }

        System.out.print("Before: ");
        for(int i : array){
            System.out.printf("%d,", i);
        }

        for(int i = 0; i < array.length; i++){
            int min = i; 
            for(int j = i+1; j < array.length; j++){
                if (array[min]>array[j]){
                    min = j;
                }
            }  
            if(min!=i){
                int temp = array[i];
                array[i] = array[min];
                array[min] = temp;
            }      
        }

        System.out.print("After: ");
        for(int i : array){
            System.out.printf("%d,", i);
        }
        
    }
}